src/otx/algo/instance_segmentation/maskrcnn.py

```diff
@@ -5,111 +5,56 @@
+from otx.algo.instance_segmentation.losses import ROICriterion, RPNCriterion
 from otx.algo.instance_segmentation.necks import FPN
-from otx.algo.instance_segmentation.two_stage import TwoStageDetector
+from otx.algo.instance_segmentation.segmentors.two_stage import TwoStageDetector
 from otx.algo.instance_segmentation.utils.roi_extractors import SingleRoIExtractor
-from otx.algo.modules.norm import build_norm_layer
 from otx.algo.utils.support_otx_v1 import OTXv1Helper
-from otx.core.config.data import TileConfig
 from otx.core.exporter.base import OTXModelExporter
 from otx.core.exporter.native import OTXNativeModelExporter
-from otx.core.metrics.mean_ap import MaskRLEMeanAPFMeasureCallable
-from otx.core.model.base import DefaultOptimizerCallable, DefaultSchedulerCallable
 from otx.core.model.instance_segmentation import ExplainableOTXInstanceSegModel
 
-if TYPE_CHECKING:
-    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
-
-    from otx.core.metrics import MetricCallable
-    from otx.core.schedulers import LRSchedulerListCallable
-    from otx.core.types.label import LabelInfoTypes
-
 
 class MaskRCNN(ExplainableOTXInstanceSegModel):
     """MaskRCNN Model."""
 
-    @property
-    def _exporter(self) -> OTXModelExporter:
-        """Creates OTXModelExporter object that can export the model."""
-        if self.input_size is None:
-            msg = f"Input size attribute is not set for {self.__class__}"
-            raise ValueError(msg)
-
-        return OTXNativeModelExporter(
-            task_level_export_parameters=self._export_parameters,
-            input_size=(1, 3, *self.input_size),
-            mean=self.mean,
-            std=self.std,
-            resize_mode="fit_to_window",
-            pad_value=0,
-            swap_rgb=False,
-            via_onnx=True,
-            onnx_export_configuration={
-                "input_names": ["image"],
-                "output_names": ["boxes", "labels", "masks"],
-                "dynamic_axes": {
-                    "image": {0: "batch", 2: "height", 3: "width"},
-                    "boxes": {0: "batch", 1: "num_dets"},
-                    "labels": {0: "batch", 1: "num_dets"},
-                    "masks": {0: "batch", 1: "num_dets", 2: "height", 3: "width"},
-                },
-                "opset_version": 11,
-                "autograd_inlining": False,
-            },
-            output_names=["bboxes", "labels", "masks"],
-        )
-
-    def load_from_otx_v1_ckpt(self, state_dict: dict, add_prefix: str = "model.") -> dict:
-        """Load the previous OTX ckpt according to OTX2.0."""
-        return OTXv1Helper.load_iseg_ckpt(state_dict, add_prefix)
-
-
-class MaskRCNNResNet50(MaskRCNN):
-    """MaskRCNN with ResNet50 backbone."""
+    AVAILABLE_MODEL_VERSIONS: ClassVar[list[str]] = [
+        "maskrcnn_resnet_50",
+        "maskrcnn_efficientnet_b2b",
+        "maskrcnn_swin_tiny",
```

Comment:
It is about the name alignment across our tasks. I can explain why I chose full name. For some tasks, we could face a problem where we share some modules and few models has small/medium/large versions. Also, for DINOV2 it is required to have full name to download model. Next, for HuggingFace wrapper we should also provide full name.

Reply:
I think maintaining this policy seems better.

Reply:
I understand. Actually, what I want to say is that it looks weird that we should pass "maskrcnn_resnet_50" as `model_name` argument to `MaskRCNN` class. So, I think it may be better that user passes "resnet_50" as model_name and `MaskRCNN` internally passes "maskrcnn_resnet_50" to common modules or other factory functions. Let's discuss it later as sungchul said.